pkg/util/varz/varz.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package varz
```

Comment:
I think varz is probably a bad name as that seems to be associated with metrics. I think statez or something is probably a better name. This is not intended for metrics, it's intended for generic application state. If that's a resonable usage of prometheus I'd be happy to move this over. From what I could tell, prometheus was more about exporting metrics (gauges, counters...).